src/LMTR_alg.jl

```diff
@@ -70,7 +70,7 @@ function LMTR(
   β = options.β
 
   local l_bound, u_bound
-  if has_bounds(nls)
+  if (has_bounds(nls) || subsolver == TRDH)
```

Comment:
Could you add a new bool variable `treats_bounds = has_bounds(nls) || subsolver == TRDH`?